extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java

```diff
@@ -222,44 +222,44 @@ public void build(
+        final AnnotationInstance applicationPath;
         final Set<String> excludedClasses;
         if (resteasyConfig.buildTimeConditionAware) {
             excludedClasses = getExcludedClasses(buildTimeConditions);
         } else {
             excludedClasses = Collections.emptySet();
         }
+        final Set<String> allowedClasses;
         final String appClass;
         if (resteasyConfig.ignoreApplicationClasses) {
+            applicationPath = null;
             allowedClasses = Collections.emptySet();
             appClass = null;
         } else {
-            applicationPaths = index.getAnnotations(ResteasyDotNames.APPLICATION_PATH);
-            allowedClasses = getAllowedClasses(index);
-            jaxrsProvidersToRegisterBuildItem = getFilteredJaxrsProvidersToRegisterBuildItem(
-                    jaxrsProvidersToRegisterBuildItem, allowedClasses, excludedClasses);
-
-            Collection<ClassInfo> knownApplications = index.getAllKnownSubclasses(ResteasyDotNames.APPLICATION).stream()
+            Collection<ClassInfo> jakartaRestApplicationClasses = index.getAllKnownSubclasses(ResteasyDotNames.APPLICATION)
+                    .stream()
                     .filter(ci -> !ci.isAbstract()).collect(
                             Collectors.toSet());
-            // getAllowedClasses throws an Exception if multiple Applications are found, so we should only get 1
-            if (knownApplications.size() == 1) {
-                appClass = knownApplications.iterator().next().name().toString();
-            } else {
+            if (jakartaRestApplicationClasses.size() > 1) {
+                throw new RuntimeException("More than one Application class: " + jakartaRestApplicationClasses);
+            }
+            if (jakartaRestApplicationClasses.isEmpty()) {
+                applicationPath = null;
+                allowedClasses = Collections.emptySet();
                 appClass = null;
+            } else {
+                ClassInfo jakartaRestApplicationClass = jakartaRestApplicationClasses.iterator().next();
+                applicationPath = jakartaRestApplicationClass.annotation(ResteasyDotNames.APPLICATION_PATH);
+                allowedClasses = getAllowedClasses(jakartaRestApplicationClass);
+                appClass = jakartaRestApplicationClass.name().toString();
             }
+
+            jaxrsProvidersToRegisterBuildItem = getFilteredJaxrsProvidersToRegisterBuildItem(
+                    jaxrsProvidersToRegisterBuildItem, allowedClasses, excludedClasses);
         }
 
         boolean filterClasses = !allowedClasses.isEmpty() || !excludedClasses.isEmpty();
 
-        // currently we only examine the first class that is annotated with @ApplicationPath so best
-        // fail if the user code has multiple such annotations instead of surprising the user
-        // at runtime
-        if (applicationPaths.size() > 1) {
-            throw createMultipleApplicationsException(applicationPaths);
```

Comment:
`createMultipleApplicationsException` is not called anymore, should be also dropped

Reply:
Good catch! It's now fixed